workflows/rnaseq/downstream/functional-enrichment.Rmd

```diff
@@ -61,69 +66,130 @@ for(comp in names(genes.df)){
 
     # loop over selection list
     for(sel in sel.names){
-
         all.enrich[[comp]][[sel]] <- list()
         enrich.files[[comp]][[sel]] <- list()
 
-        gene <- genes.df[[comp]][[sel]][[go.keytype]]
+        if(analysis.mode == 'gsea'){
+            # for 'gsea' we create a decreasing ranked named vector of values
+
+            # named vector with ID -> entrezID mapping
+            idmap <- genes.df[[comp]][[sel]][[go.keytype]]
+            names(idmap) <- genes.df[[comp]][[sel]][[from.type]]
+    
+            # we use the 'stat' column as the ranking metric
```

Comment:
I think I get the argument for using `stat` (it reflects pval and already has a sign) but what about log2FoldChange? The argument from DESeq2 vignette/paper is that shrunken LFC is good for ranking.

Reply:
That's a good idea, it will be worth testing this out